### DISCUSSION TITLE
Root certificate

### DISCUSSION THREAD

**Discussion by alanhkarp** [ucan-wg/spec#109] [Ideas]
[Section 6.2](https://github.com/ucan-wg/spec/#62-principal-alignment) of the spec shows an example where the root certificate for a resource is signed by the resource owner.  That means that the verifier will have to keep track of the public keys of all the owners.  That job would be simpler if the root certificate were created by the resource server  and the user got a delegation of it.  Then the verifier would only need to track the public keys of the resource servers.  

An added benefit is that there may be administrative functions at the resource server, e.g., audit, performance monitoring, that should be granted to a system administrator but not the owner.  The root certificate can contain both sets of permissions, with one subset being delegated to the user and the other to an administrator.

One way to look at it is who has a "contract" with whom.  The verifier and the resource server have one, and the user and resource server have another one.  Note that there is no such relationship between the user and the verifier.  What is currently in Section 6.2 obscures this fact.

**Comment by expede** [comment]
@alanhkarp Ah! Perhaps to clarify: in a lot of cases, we don't have a "resource server", for example my writing the DID to a `.well-known` path inside a data structure. This means that one DID may control `n` resources, but they're self verifying.

If that makes sense, I can reword to make it clearer!

**Reply by expede** [reply]
(Also, I'd still love to schedule some time to pick your brain about a couple things. My schedule is pretty wild the next ~2.5 weeks, let's find some time afterwords!)

**Comment by alanhkarp** [comment]
I don't understand what "self verifying" means.  Some entity, what I called
the verifier, has to make sure that the delegations are valid.

That being said, whoever creates the resource produces the root UCAN.  That
may be you in your use case, but things are simpler if there is a resource
server and it creates the root UCAN.

--------------
Alan Karp

**Comment by alanhkarp** [comment]
My calendar is anything but wild, just the way I like it.  I can usually
make time on short notice if one of your meetings gets canceled.  The only
time I'm not free is Friday mornings 10-12 Pacific when our capability
group meets.

--------------
Alan Karp
